andela/activo-api
# #157505109 Add SQLAlchemy Hook

#### What does this PR do?
Adds and configures `flask-sqlalchemy` for the project.

#### Description of Task to be completed?
##### Why this is needed
Activo will use SQLAlchemy as the ORM tool and this chore adds it to the project

##### Acceptance Criteria
The Activo api project having SQLAlchemy added with one model (user) and an initial database upgrade script. 

##### Dev Notes
The database to connect here is PostgreSQL
User model attributes
-name
-email
-image_url
The `user` model should be defined inside the `models` folder. 

#### How should this be manually tested?
Some configuration to be done:
1. Create a `.env` file in your project root
2. Add the following to the `.env` file:
```
FLASK_ENV=development
SQLALCHEMY_DATABASE_URI=postgresql://{your db username}:{your db password}@localhost/{your db name}'
```
3. Note that the `FLASK_ENV` variable can take any one of the values `development`, `testing` or `production` and will load the corresponding configuration as set in the `config.py` file.
4. Run migrations by executing the command `flask db upgrade`.
5. At this point, the app can be run by executing the command `python manage.py`.
6. Open your browser and navigate to `127.0.0.1:5000`. You should see the text `{message="Welcome to the AM api"}`

#### Any background context you want to provide?
Switched from `flask-script` to the `flask-cli` tools pre-bundled with flask.

#### What are the relevant pivotal tracker stories?
[#157505109](https://www.pivotaltracker.com/story/show/157505109)

#### Screenshots (if appropriate)
N/A

#### Questions:
None
